### PR TITLE
Update pagerduty modules to rest v2

### PR DESCRIPTION
**Pull request by alex-dukhno**
##### SUMMARY
This PR updates modules that work with PagerDuty REST API from version 1 to version 2.
The v1 is in the [process of decommission](https://v2.developer.pagerduty.com/v2/docs/v1-rest-api-decommissioning-faq) and ansible modules have to be updated to v2 to continue work with PagerDuty.
Users that have api v1 token can still use it till the end of decommissioning process, which will end on ```October 19, 2018, at 22:00 PDT (UTC+7)```

The PR:
 * contains structural changes of `pagerduty` module, it's made more OOP stylish
 * introduces changes to work with PagerDuty REST API v2 and support for v1 tokens
 * adds unit tests for `pagerduty` and `pagerduty_alerts` modules
 * updates documentation to both modules

##### ISSUE TYPE
 - Feature Pull Request
 - Docs Pull Request

##### COMPONENT NAME
```pagerduty``` and ```pagerduty_alerts``` modules

##### ANSIBLE VERSION
```
ansible 2.7.0.dev0 (update-pagerduty-modules-to-REST-v2 cea693b000) last updated 2018/07/11 14:07:07 (GMT +300)
  config file = /Users/adukhno/.ansible.cfg
  configured module search path = [u'/Users/adukhno/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/adukhno/Workspace/ansible/lib/ansible
  executable location = /Users/adukhno/Workspace/ansible/bin/ansible
  python version = 2.7.15 (default, Jun 17 2018, 12:46:58) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
```

### CODE REVIEW

File: lib/ansible/modules/monitoring/pagerduty_alert.py [hunk 1]
```diff
@@ -26,13 +26,18 @@
...
     name:
         description:
-            - PagerDuty unique subdomain.
+            - PagerDuty unique subdomain. Obsolete. It is not used with PagerDuty REST v2 API.
+    service_id:
+        description:
+            - ID of PagerDuty service when incidents will be triggered, acknowledged or resolved.
         required: true
-    service_key:
+        version_added: '2.7.0'
```
Comment (gundalow) [inline lib/ansible/modules/monitoring/pagerduty_alert.py]: `version_added: "2.7"`

File: lib/ansible/modules/monitoring/pagerduty_alert.py [hunk 2]
```diff
@@ -26,13 +26,18 @@
...
+    integration_key:
         description:
             - The GUID of one of your "Generic API" services.
-            - This is the "service key" listed on a Generic API's service detail page.
+            - This is the "integration key" listed on a "Integrations" tab of PagerDuty service.
         required: true
+        version_added: '2.7.0'
```
Comment (gundalow) [inline lib/ansible/modules/monitoring/pagerduty_alert.py]: `version_added: "2.7"`

File: lib/ansible/modules/monitoring/pagerduty_alert.py [hunk 3]
```diff
@@ -62,6 +67,7 @@
             - For C(acknowledged) or C(resolved) I(state) - This should be the incident_key you received back when the incident was first opened by a
               trigger event. Acknowledge events referencing resolved or nonexistent incidents will be discarded.
         required: false
+        version_added: '2.7.0'
```
Comment (gundalow) [inline lib/ansible/modules/monitoring/pagerduty_alert.py]: `version_added: "2.7"`

File: lib/ansible/modules/monitoring/pagerduty.py [hunk 4]
```diff
@@ -35,31 +35,27 @@
...
+            - PagerDuty unique subdomain. Obsolete. It is not used with PagerDuty REST v2 API.
     user:
         description:
-            - PagerDuty user ID.
-        required: true
-    passwd:
-        description:
-            - PagerDuty user password.
-        required: true
+            - PagerDuty user ID. Obsolete. Please, use token for authorization.
```
Comment (gundalow) [inline lib/ansible/modules/monitoring/pagerduty.py]: `PagerDuty user ID. Obsolete. Please, use I(token) for authorization.`

**Comment by alex-dukhno** [comment]
@gundalow, I addressed you review comments. Can you check, please?

**Comment by alex-dukhno** [comment]
@gundalow, is there anything I need to do that PR could be accepted?

**Comment by alex-dukhno** [comment]
ping @ryansb , @gundalow

File: lib/ansible/modules/monitoring/pagerduty_alert.py [hunk 5]
```diff
@@ -26,13 +26,18 @@
 options:
     name:
         description:
-            - PagerDuty unique subdomain.
+            - PagerDuty unique subdomain. Obsolete. It is not used with PagerDuty REST v2 API.
+    service_id:
+        description:
+            - ID of PagerDuty service when incidents will be triggered, acknowledged or resolved.
         required: true
-    service_key:
```
Comment (ryansb) [inline lib/ansible/modules/monitoring/pagerduty_alert.py]: Is service key no longer a parameter that's accepted? To help with backwards compatibility, maybe we could add this as an alias of `integration_key`

**Review by gundalow — APPROVED** [review]
Once aliases has been added as suggested by Ryan

**Review by ryansb — APPROVED** [review]

**Comment by ryansb** [comment]
Thanks @Alex-Diez for your patience in doing this rewrite, and for adding tests as you go!

**Comment by craigcalef** [comment]
Tested with ansible 2.6.4 and it was able to create a maintenance window successfully.

**Pull request merged**